source/particle/interpolator/cell_average.cc

```diff
@@ -74,8 +75,10 @@ namespace aspect
                 ghost_particles.equal_range(cell_index);
 
         const unsigned int n_particles = std::distance(particle_range.first,particle_range.second);
-        const unsigned int n_properties = particles.begin()->second.get_properties().size();
-        std::vector<double> cell_properties (n_properties,0.0);
+        const unsigned int n_particle_properties = particles.begin()->second.get_properties().size();
+        const unsigned int n_selected_properties = selected_properties.n_selected_components(n_particle_properties);
+
+        std::vector<double> cell_properties (n_selected_properties,0.0);
```

Comment:
This answers my question above, but I would prefer if the size of the vector was actually `n_particle_properties`.

You may also want to initialize with `numbers::signaling_NaN<double>()` in that case to poison elements you don't touch.